engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformulamulticolumn/RollingFormulaMultiColumnOperator.java

```diff
@@ -0,0 +1,394 @@
+            // Assign the influencer values chunks to the input consumers.
+            for (int i = 0; i < valueChunks.length; i++) {
+                inputConsumers[i].setInfluencerValuesChunk(valueChunks[i]);
+            }
+        }
+
+        @Override
+        protected void push(int pos, int count) {
+            throw new IllegalStateException("RollingFormulaMultiColumnOperator.Context.push should never be called.");
+        }
+
+        @Override
+        public void accumulateCumulative(
+                @NotNull final RowSequence inputKeys,
+                @NotNull final Chunk<? extends Values>[] valueChunkArr,
+                @Nullable final LongChunk<? extends Values> tsChunk,
+                final int len) {
+            throw new UnsupportedOperationException("RollingFormula is not supported in cumulative operations.");
+        }
+
+        @Override
+        public void accumulateRolling(
+                @NotNull final RowSequence inputKeys,
+                @NotNull final Chunk<? extends Values>[] influencerValueChunkArr,
+                @Nullable final LongChunk<OrderedRowKeys> affectedPosChunk,
+                @Nullable final LongChunk<OrderedRowKeys> influencerPosChunk,
+                @NotNull final IntChunk<? extends Values> pushChunk,
+                @NotNull final IntChunk<? extends Values> popChunk,
+                final int len) {
+
+            // TODO: reuse a
+            setValueChunks(influencerValueChunkArr);
+            setPosChunks(affectedPosChunk, influencerPosChunk);
+
+            int pushIndex = 0;
+
+            // chunk processing
+            for (int ii = 0; ii < len; ii++) {
+                final int pushCount = pushChunk.get(ii);
+                final int popCount = popChunk.get(ii);
+
+                if (pushCount == NULL_INT) {
+                    outputValues.fillWithNullValue(ii, 1);
+                    continue;
+                }
+
+                // pop for this row
+                if (popCount > 0) {
+                    for (RingBufferWindowConsumer consumer : inputConsumers) {
+                        consumer.pop(popCount);
+                    }
+                }
+
+                // push for this row
+                if (pushCount > 0) {
+                    for (RingBufferWindowConsumer consumer : inputConsumers) {
+                        consumer.push(pushIndex, pushCount);
+                    }
+                    pushIndex += pushCount;
+                }
+
+                // If not empty (even if completely full of null), run the formula over the window values.
+                outputSetter.accept(ii);
+            }
+
+            // chunk output to column
+            writeToOutputColumn(inputKeys);
+        }
+
+        @Override
+        protected void writeToOutputChunk(int outIdx) {
+            throw Assert.statementNeverExecuted("RollingFormulaMultiColumnOperator.Context.writeToOutputChunk");
```

Comment:
This is so weird to me. We are throwing something that returns null, assuming we will never get here since it will internally throw something.
Also we are using it all over the code. Maybe we can keep just the Assert part
```suggestion
           Assert.statementNeverExecuted("RollingFormulaMultiColumnOperator.Context.writeToOutputChunk");
```

Or maybe an IllegalStateException like you used in line 119